### PR TITLE
Dart support

### PR DESCRIPTION
**Pull request by g5becks**
This pr add full support (client and server) for [Dart](https://dart.dev/) programming language to webrpc as requested here #67.

### CODE REVIEW

File: _examples/dart_full_stack/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (pkieltyka) [inline _examples/dart_full_stack/.vscode/launch.json:1]: please remove this file from the commit, it shouldn't be part of version control

**Comment by pkieltyka** [comment]
thanks @g5becks that is quite a substantial bit of work and example, this is awesome

**Comment by g5becks** [comment]
@pkieltyka No problem, thanks for the great framework.

**Comment by g5becks** [comment]
@pkieltyka Is there an issue preventing this from being merged? I want to add C# support as well but I want to make sure that this work is accepted first.

Thanks.

File: gen/golang/embed/statik.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Code generated by statik. DO NOT EDIT.
+
+package embed
+
+import (
+	"github.com/rakyll/statik/fs"
```
Comment (pkieltyka) [inline gen/golang/embed/statik.go:6]: @g5becks in webrpc, we're using a forked version of statik at github.com/goware/statik. There are a number of improvements in goware/statik that are required, and in this case.. this file wouldn't need to exist

File: go.mod [hunk 3]
```diff
@@ -3,11 +3,11 @@ module github.com/webrpc/webrpc
 go 1.13
 
 require (
-	github.com/go-chi/chi v4.0.2+incompatible
+	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/goware/statik v0.2.0
 	github.com/pkg/errors v0.8.1
-	github.com/stretchr/objx v0.2.0 // indirect
+	github.com/rakyll/statik v0.1.7
```
Comment (pkieltyka) [inline go.mod:9]: should be using goware/statik

File: _examples/dart_full_stack/Taskfile.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+version: "2"
```
Comment (pkieltyka) [inline _examples/dart_full_stack/Taskfile.yml:1]: Would you be cool to move this dart_full_stack example to another repo such as "flutter-webrpc-example", even under your own repo and we can link to it from README? 

as well, inside of webrpc repo, under _examples/ we could make a "dart-app" which includes a simple client and server example, ideally something very minimal

Comment (g5becks) [reply _examples/dart_full_stack/Taskfile.yml:1]: I don't have a problem with that. I'm not sure how much more minimal the example could be though. I think most of the complexity is inherent in even the smallest of flutter apps to be honest?. Do you mean fewer methods in the service?

Comment (g5becks) [reply _examples/dart_full_stack/Taskfile.yml:1]: or would you like me to forgo using flutter altogether?

Comment (pkieltyka) [reply _examples/dart_full_stack/Taskfile.yml:1]: yes, for the example in this repo specifically, the examples should be as minimal/small as possible.

Comment (pkieltyka) [reply _examples/dart_full_stack/Taskfile.yml:1]: remove flutter, yes

Comment (pkieltyka) [reply _examples/dart_full_stack/Taskfile.yml:1]: but I still think the flutter example is great and would be a nice reference in another example repo.. up to you for that of course

Comment (g5becks) [reply _examples/dart_full_stack/Taskfile.yml:1]: No problem, I will write a quick example over the weekend and move the current one to another repo.

Comment (g5becks) [reply _examples/dart_full_stack/Taskfile.yml]: Hey @pkieltyka , sorry it took me so long to update the examples for this. I've been slammed at work. Anyhow, I have moved the flutter examples to another repo and created simpler example for dart usage. Please let me know if there's anything else needed.

File: gen/dart/funcmap.go [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+	return in == "flutter_bloc"
+}
+
+func useAngularBloc(in string) bool {
+	return in == "angular_bloc"
+}
+
+func useBloc(in string) bool {
+	return in == "angular_bloc" || in == "flutter_bloc"
+}
```
Comment (VojtechVitek) [inline gen/dart/funcmap.go:253]: I'm reading through these template functions for #110.

**Pull request closed without merging**